sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -164,6 +164,7 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper {
     case _: BinaryComparison => true
     case _: In | _: InSet => true
     case _: StringPredicate => true
+    case _: MultiLikeBase => true
```

Comment:
so this is a regression in 3.1 because we introduced this `MultiLikeBase`?

Reply:
No. we will simplify `MultiLikeBase` by SPARK-33938.

Reply:
the fixed version of SPARK-33938 is 3.1.1, so it's a 3.1 regression?

Reply:
Ah, yes. I forget 3.1.0.